src/transaction.js

```diff
@@ -125,11 +125,18 @@ function Transaction(session, options) {
    */
   this.transaction = true;
 
+  options = options || {};
+
   this.queuedMutations_ = [];
   this.runFn_ = null;
 
-  this.timeout_ = DEFAULT_TRANSACTION_TIMEOUT;
   this.beginTime_ = null;
+  this.timeout_ = DEFAULT_TRANSACTION_TIMEOUT;
+
+  if (options.timeout) {
+    this.timeout_ = options.timeout;
+    delete options.timeout;
```

Comment:
We should probably not delete the user-held property (unless `options` is a cloned object already?)

Reply:
Good call, I've made an update PTAL